langchain/chains/api/base.py

```diff
@@ -0,0 +1,45 @@
+from langchain import LLMChain
+from typing import Optional, List
+from pydantic import BaseModel
+from langchain.chains.base import Chain
+from langchain.chains.api.prompt import API_URL_PROMPT, API_RESPONSE_PROMPT
+from langchain.llms.base import LLM
+import requests
+
+class RequestsChain(BaseModel):
+    headers : Optional[dict] = None
+
+    def run(self, url: str) -> str:
+      return requests.get(url, headers=self.headers).text
+
+class APIChain(Chain, BaseModel):
+  api_request_chain: LLMChain
+  api_answer_chain: LLMChain
+  requests_chain: RequestsChain
+  api_docs: str
+  question_key: str = "question"
+  output_key : str = "output"
+  
+  @property
+  def input_keys(self) -> List[str]:
+    """Input keys this chain expects."""
+    return [self.question_key]
+
+  @property
+  def output_keys(self) -> List[str]:
+    """Output keys this chain expects."""
+    return  [self.output_key]
+
+  def _call(self, inputs):
+      question = inputs[self.question_key]
+      api_url = self.api_request_chain.predict(question=question,api_docs=self.api_docs)
+      api_response = self.requests_chain.run(api_url)
+      answer = self.api_answer_chain.predict(question=question,api_docs=self.api_docs, api_url=api_url, api_response=api_response)
+      return {self.output_key: answer}
+
+  @classmethod
+  def from_llm_and_api_docs(cls, llm: LLM, api_docs: str):
```

Comment:
i would add **kwargs here in case they want to specify other things like verbose-True or somethign

Reply:
Added!